core/src/test/java/org/elasticsearch/action/get/GetResponseTests.java

```diff
@@ -0,0 +1,357 @@
+            }
+            if (randomBoolean() && currentDepth < 5) {
+                builder.startObject(randomAsciiOfLengthBetween(3, 10));
+                addFields(builder, currentDepth + 1);
+                builder.endObject();
+            }
+            if (randomBoolean() && currentDepth < 5) {
+                builder.startArray(randomAsciiOfLengthBetween(3, 10));
+                int numElements = randomIntBetween(1, 5);
+                for (int j = 0; j < numElements; j++) {
+                    if (frequently()) {
+                        builder.value(randomBoolean() ? randomLong() : randomAsciiOfLengthBetween(3, 10));
+                    } else {
+                        builder.startObject();
+                        addFields(builder, currentDepth + 1);
+                        builder.endObject();
+                    }
+                }
+                builder.endArray();
+            }
+        }
+    }
+
+    private static Tuple<Map<String, GetField>,Map<String, GetField>> randomGetFields() {
+        int numFields = randomIntBetween(1, 10);
+        Map<String, GetField> fields = new HashMap<>(numFields);
+        Map<String, GetField> expectedFields = new HashMap<>(numFields);
+        for (int i = 0; i < numFields; i++) {
+            Tuple<GetField, GetField> tuple = randomGetField();
+            GetField getField = tuple.v1();
+            GetField expectedGetField = tuple.v2();
+            fields.put(getField.getName(), getField);
+            expectedFields.put(expectedGetField.getName(), expectedGetField);
+        }
+        return Tuple.tuple(fields, expectedFields);
+    }
+
+    private static Tuple<GetField, GetField> randomGetField() {
+        if (randomBoolean()) {
+            String fieldName = randomFrom(ParentFieldMapper.NAME, RoutingFieldMapper.NAME, UidFieldMapper.NAME);
+            GetField getField = new GetField(fieldName, Collections.singletonList(randomAsciiOfLengthBetween(3, 10)));
+            return Tuple.tuple(getField, getField);
+        }
+        String fieldName = randomAsciiOfLengthBetween(3, 10);
+        int numValues = randomIntBetween(1, 5);
+        List<Object> values = new ArrayList<>();
+        List<Object> expectedValues = new ArrayList<>();
+        int dataType = randomIntBetween(0, 8);
+        for (int i = 0; i < numValues; i++) {
+            switch(dataType) {
+                case 0:
+                    long randomLong = randomLong();
+                    values.add(randomLong);
+                    expectedValues.add(randomLong);
+                    break;
+                case 1:
+                    int randomInt = randomInt();
+                    values.add(randomInt);
+                    expectedValues.add(randomInt);
+                    break;
+                case 2:
+                    Short randomShort = randomShort();
+                    values.add(randomShort);
+                    expectedValues.add(randomShort.intValue());
```

Comment:
that wouldn't quite work. The list is typed `<Object>` if I add `randomShort` the `short` becomes `Short`, but I need an `Integer` in the map.